app/controllers/api/school_students_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Api
+  class SchoolStudentsController < ApiController
+    before_action :authorize_user
+    load_and_authorize_resource :school
+    authorize_resource :school_student, class: false
+
+    def index
+      result = SchoolStudent::List.call(school: @school, token: current_user.token)
+
+      if result.success?
+        @school_students = result[:school_students]
+        render :index, formats: [:json], status: :ok
+      else
+        render json: { error: result[:error] }, status: :unprocessable_entity
+      end
+    end
+
+    def create
+      result = SchoolStudent::Create.call(school: @school, school_student_params:, token: current_user.token)
+
+      if result.success?
+        head :no_content
+      else
+        render json: { error: result[:error] }, status: :unprocessable_entity
+      end
+    end
+
+    def create_batch
+      result = SchoolStudent::CreateBatch.call(school: @school, uploaded_file: params[:file], token: current_user.token)
+
+      if result.success?
+        head :no_content
+      else
+        render json: { error: result[:error] }, status: :unprocessable_entity
+      end
+    end
+
+    def update
+      result = SchoolStudent::Update.call(school: @school, school_student_params:, token: current_user.token)
+
+      if result.success?
+        head :no_content
+      else
+        render json: { error: result[:error] }, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      result = SchoolStudent::Delete.call(school: @school, student_id: params[:id], token: current_user.token)
+
+      if result.success?
+        head :no_content
+      else
+        render json: { error: result[:error] }, status: :unprocessable_entity
+      end
+    end
```

Comment:
🔧 update destroy actions to anonymise and/or archive. We need to discuss and agree exactly what needs altering but school student IDs need to remain to enable us to have a trail in our app